test/gtest/uct/uct_test.cc

```diff
@@ -160,45 +217,59 @@ std::vector<const resource*> uct_test::enum_resources(const std::string& tl_name
         ASSERT_UCS_OK(status);
 
         for (unsigned i = 0; i < num_md_resources; ++i) {
-            uct_md_h pd;
+            uct_md_h md;
             uct_md_config_t *md_config;
             status = uct_md_config_read(md_resources[i].md_name, NULL, NULL,
                                         &md_config);
             ASSERT_UCS_OK(status);
 
             {
                 scoped_log_handler slh(hide_errors_logger);
-                status = uct_md_open(md_resources[i].md_name, md_config, &pd);
+                status = uct_md_open(md_resources[i].md_name, md_config, &md);
             }
             uct_config_release(md_config);
             if (status != UCS_OK) {
                 continue;
             }
 
             uct_md_attr_t md_attr;
-            status = uct_md_query(pd, &md_attr);
+            status = uct_md_query(md, &md_attr);
             ASSERT_UCS_OK(status);
 
-            status = uct_md_query_tl_resources(pd, &tl_resources, &num_tl_resources);
+            status = uct_md_query_tl_resources(md, &tl_resources, &num_tl_resources);
             ASSERT_UCS_OK(status);
 
+            resource_speed tcp_fastest_rsc;
+
             for (unsigned j = 0; j < num_tl_resources; ++j) {
-                resource rsc;
-                rsc.md_name    = md_resources[i].md_name;
-                rsc.local_cpus = md_attr.local_cpus;
-                rsc.tl_name    = tl_resources[j].tl_name;
-                rsc.dev_name   = tl_resources[j].dev_name;
-                rsc.dev_type   = tl_resources[j].dev_type;
-                all_resources.push_back(rsc);
+                if (tcp_fastest_dev && (std::string("tcp") == tl_resources[j].tl_name)) {
+                    resource_speed rsc(md, std::string(md_resources[i].md_name), md_attr.local_cpus,
```

Comment:
maybe we can add resource constructor from: (&md_resources[i], &md_attr, &tl_resources[j])?